sdk-cpp/include/do_errors.h

```diff
@@ -29,22 +29,52 @@ enum class errc : int32_t
     do_e_unknown_property_id    = -2133843951
 };
 
-#if (DO_ENABLE_EXCEPTIONS)
-
+// Error category can denote different types of error - may be useful for categorizing whether error code is DO error/posix error/std::error
+// Problem is it has a deleted copy constructor which makes using it
 class error_category : public std::error_category
 {
 public:
     const char* name() const noexcept override;
 
-    std::string message(int code) const override;
+    std::string message(int32_t code) const override;
+};
+
+class error_code : public std::error_code
+{
+public:
+    error_code() = default;
+    
+    error_code(int32_t code) : _code(code)
+    {
+        
+    }
+
+    error_code(errc code)
+    {
+        _code = static_cast<int32_t>(code);
+    }
+
+    int32_t value() const noexcept
+    {
+        return _code;
+    }
+
+    /*
+    const error_category& category() const noexcept
+    {
+        return _category;
+    }
+    */
+
+private:
+    int32_t _code;
+    //error_category _category;
```

Comment:
>error_category [](http://example.com/codeflow?start=6&length=14)

IIRC, this should be a reference. Commonly, a reference to a static instance. See the constructor here: [https://en.cppreference.com/w/cpp/error/error_code/error_code](https://en.cppreference.com/w/cpp/error/error_code/error_code)

and cpprest's category here:
[https://github.com/microsoft/cpprestsdk/blob/master/Release/src/utilities/asyncrt_utils.cpp](https://github.com/microsoft/cpprestsdk/blob/master/Release/src/utilities/asyncrt_utils.cpp)

Reply:
sounds good to me